data_analysis/data_capture.py

```diff
@@ -16,25 +16,31 @@
 
 from data_types import Recording, RecordingEnvironment, Event
 
-
 # Name of your serial port (e.g., '/dev/cu.usbserial-0001', or COM5).
-def collect_data(seconds: float, serial_port = '/dev/cu.usbserial-0001'):
+def collect_data(seconds: float, serial_port = '/dev/cu.usbserial-0001', fs=None, error_threshold=0.5):
     # Open the serial port for communication with the ESP8266.
     ser = serial.Serial(serial_port, 115200)
     t_end = time.time() + seconds # Data collection will run for this many seconds
     vibes = [] # Raw accelerometer data
     button_presses = [] # Timestamps of button presses
     start_time = None
+    last_time = None
 
     try:
         while time.time() < t_end:
             timestamp, accel, event = ser.readline().decode().strip().split(',') # Read data from the serial port.
```

Comment:
We should catch any unparsable serial input and raise an error here. Wrap this in a try-except block and raise an error showing the input that caused it

Reply:
adding this in new commit:
`try:
                timestamp, accel, event = ser.readline().decode().strip().split(',') # Read data from the serial port.
            except ValueError:
                print('ValueError:', ser.readline().decode().strip())`

Reply:
updated to raise error